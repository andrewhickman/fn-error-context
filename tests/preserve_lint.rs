// See https://github.com/andrewhickman/fn-error-context/issues/3
#![deny(dead_code)]

use fn_error_context::context;

#[context("context")]
fn do_stuff() -> anyhow::Result<()> {
    anyhow::bail!("error")
}

fn main() {}
