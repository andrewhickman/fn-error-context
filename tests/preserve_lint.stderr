error: function is never used: `do_stuff`
 --> $DIR/preserve_lint.rs:7:4
  |
7 | fn do_stuff() -> anyhow::Result<()> {
  |    ^^^^^^^^
  |
note: the lint level is defined here
 --> $DIR/preserve_lint.rs:2:9
  |
2 | #![deny(dead_code)]
  |         ^^^^^^^^^
